src/webapp/reports/autogenerated-forms/GridWithPeriodsViewModel.ts

```diff
@@ -213,6 +204,31 @@ export class GridWithPeriodsViewModel {
         };
     }
 
+    private static getColumnWithDataElements(selectedDataElements: DataElement[], column: string): TotalItem[] {
```

Comment:
Is this almost the same method that we have in `GridFormViewModel.ts`, right? We're only using the column param for the error message, so I think we can reuse this method in both `GridFormViewModel.ts` and `GridWithPeriodsViewModel`

```ts
private static getColumnWithDataElements(selectedDataElements: DataElement[], columnName: string): TotalItem[] {
   // rest of the code
   
   if (!categoryOptionCombo)
       throw new Error(
         `Cannot found categoryOptionCombo in column ${columnName} for dataElement ${dataElement.code}`
       );
}
```

And we pass the name property instead the whole object

```ts
const columnWithDataElements = GridViewModel.getColumnWithDataElements(
  selectedDataElements,
  column.name
);
```

Does this make sense?